src/aiotaskq/worker.py

```diff
@@ -20,47 +23,170 @@ class Defaults:
+        "\tconcurrency: %s\n",
+        *(pid, "1.0.0", poll_interval_s, REDIS_URL, concurrency),
     )
 
-    redis_client = aioredis.from_url(REDIS_URL)
+    # Ensure child worker processes log to parent's stderr
+    multiprocessing.log_to_stderr(logging.DEBUG)
+
+    # Start child worker processes in background
+    child_worker_processes: list["multiprocessing.Process"] = [
+        multiprocessing.Process(target=_worker, args=(app_import_path, poll_interval_s))
+        for _ in range(concurrency)
+    ]
+    for p in child_worker_processes:
+        p.start()
+    child_worker_pids = [c.pid for c in child_worker_processes]
+
+    # Main worker accepts new task and pass it on to one of child workers
+    logger.info(
+        "[%s] Forked %s child worker processes: [pids=%s]",
+        *(pid, len(child_worker_processes), child_worker_pids),
+    )
+    redis_client: aioredis.Redis = aioredis.from_url(REDIS_URL)
     async with redis_client.pubsub() as pubsub:
         await pubsub.subscribe(TASKS_CHANNEL)
-
+        counter = 0
         while True:
             # Poll for a new task until it's available
-            result = None
-            while result is None:
-                result = await pubsub.get_message(ignore_subscribe_messages=True)
+            message: t.Optional[str] = None
+            while message is None:
+                message = await pubsub.get_message(ignore_subscribe_messages=True)
                 await asyncio.sleep(poll_interval_s)
 
             # A new task is now available
-            task_info = json.loads(result["data"])
-            task_args = task_info["args"]
-            task_kwargs = task_info["kwargs"]
-            task_id: str = task_info["task_id"]
-            task_func_name: str = task_id.split(":")[0].split(".")[-1]
-            task_func = getattr(app, task_func_name)
-
-            # Execute the task
-            logger.debug("Executing task %s(*%s, **%s)", task_id, task_args, task_kwargs)
-            task_result = task_func(*task_args, **task_kwargs)
-
-            # Publish the task return value
-            task_result = json.dumps(task_result)
-            await redis_client.publish(
-                RESULTS_CHANNEL_TEMPLATE.format(task_id=task_id), message=task_result
+            # Pass the task to one of the workers worker
+            i = counter % len(child_worker_processes)
+            selected_child_worker = child_worker_processes[i]
+            channel: str = _get_child_worker_tasks_channel(pid=selected_child_worker.pid)
+            logger.debug(
+                "[%s] Passing task to %sth child worker [message=%s, channel=%s]",
+                *(pid, i, message, channel),
             )
+            # Wait for all child workers to be ready before publishing
+            await _wait_for_child_workers_ready(
+                publisher=redis_client,
+                child_worker_pids=child_worker_pids,
+                poll_interval_s=poll_interval_s,
+            )
```

Comment:
Didn't really understand this. Why do we need to wait for all child workers to be ready before we pass a task to a child worker?

Reply:
This is a long time ago but iirc there was a case where:

> Main worker passes a task T1 to the task channel to be picked up by a child worker W1
> But W1 has not subscribed to a the channel yet
> So T1 was never picked up

So that's I wanted to wait for all child workers be ready (i.e. subscribed to the task channel) before passing any task to any child worker. But I will double check this again to be sure.

Reply:
Seems like either my memory was wrong or I misunderstood the bug and applied this logic for the wrong reason. 

This is what I did to confirm the above:
1. Remove these lines:
    ```python
    if not is_worker_ready:
            raise WorkerNotReady("No worker is ready to pick up tasks. Have you run your workers?")
     ```
     And 
     ```python
     await _wait_for_child_workers_ready(
             publisher=redis_client,
             child_worker_pids=child_worker_pids,
             poll_interval_s=poll_interval_s,
     )
     ```
2. Add `await asyncio.sleep(5)` in `worker._worker._main_loop` before subscribing
    ```python
        await asyncio.sleep(5)
        async with redis_client.pubsub() as pubsub:
     ```
3. Run `pytest -s -k test_sync_and_async_parity__simple_app`

And the test should still pass.

So I'll remove this logic.

Reply:
Done